### DISCUSSION TITLE
Bazel BZLMOD Support

### DISCUSSION THREAD

**Discussion by Attempt3035** [capnproto/capnproto#1908] [Q&A]
Hi all!
Just wondering if bzlmod support is planned or has been considered? It would be great to have this tool in the bazel central registry!
Cheers!

**Comment by kentonv** [comment]
Hmm. The current Bazel files exist basically to support [workerd](https://github.com/cloudflare/workerd) (whose developers are also the maintainers of Cap'n Proto). We haven't really thought about what is needed for these files to be consumed by third parties, much less whether we want to publish them to some sort of public repository. Generally, Cap'n Proto is provided open source in the hopes that it is useful but we don't have bandwidth to polish it as a product for third-party consumption. But I think publishing to a repository implies some basic commitment to not break things so it's something we'd want to be cautious about. We'd at the very least need to review the bazel files and make sure they are organized in the way would like third parties to consume them.

One weird thing about them right now is that everything is prefixed by `src`, e.g. `//src/capnp:capnp-rpc`... it feels kinda warty, but IIRC there were challenges to avoiding that.

cc @mikea @fhanau who do the most work maintaining these files, and so would have to work harder if they were public.

Also cc @sgammon who posted #1904 recently which adds features to the bazel files...

**Reply by sgammon** [reply]
I recently published my first Bzlmod with 'rules_graalvm'; it can be surprisingly seamless to transition.

If I can be of any help please let me know!
